File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.py
candidate = []
selected = []
visited = [0] * 9
flag = False
for _ in range(9):
    candidate.append(int(input()))


def brute_force(cnt, idx, sum):
    global flag
    if flag: return
    if cnt == 7:
        if sum == 100:
            flag = True
            selected.sort()
            print(*selected, sep="\n")
        return

    for i in range(idx, 9):
        if visited[i] != 0: continue
        visited[i] = 1
        selected.append(candidate[i])
        brute_force(cnt + 1, i + 1, sum + candidate[i])
        visited[i] = 0
        selected.pop()
        brute_force(cnt, i + 1, sum)


brute_force(0, 0, 0)

File: 백준/Bronze/2309. 일곱 난쟁이/일곱 난쟁이.cc
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>

using namespace std;


int main() {
	vector<int> lst;
	int input;
	int sum = 0;
	for (int i = 0; i < 9; i++) {
		cin >> input;
		sum += input;
		lst.push_back(input);
	}
	bool flag = 0;
	for (int i = 0; i < 9 && !flag; i++) {
		for (int j = i+1; j < 9; j++) {
			if (sum - lst[i] - lst[j] == 100) {
				lst[i] = 0;
				lst[j] = 0;
				flag= 1;
				break;
			}
		}
	}
	sort(lst.begin(), lst.end());
	for (int i = 2; i < 9; i++) {
		cout << lst[i] << endl;
	}


	return 0;
}